apis/mlops/agent/agent.proto

```diff
@@ -27,6 +27,17 @@ message ModelEventMessage {
   Event event = 5;
   string message = 6;
   uint64 availableMemoryBytes = 7;
+  ModelConfig modelConfig = 8;
+}
+
+message ModelConfig {
+  uint32 instanceCount = 1; // The number of instances of the model on a replica
+  enum Resource {
```

Comment:
I'm a bit unsure of the current structure here, mostly because in principle a Model should be able to request _both_ Memory and GPU memory, **and** different number of instances running on CPU, GPUs, etc.  For a LLM, the weights will be first be loaded into RAM and then copied into GPU memory from there (there are fancier options for DMA directly from disks or network cards into GPU memory but let's ignore those for now). This means that I need approximately the same Memory as the GPU memory requirement.

Triton is limited in its expressiveness for this to something like "give me 2 copies on the CPU, 1 copy on GPU 1 and 2 copies each on GPUs 2 and 3". MLServer is limited to "load this model n times" (where we assume the result needs n times the memory)

Given that at the moment our scheduler only considers system memory, I propose we just expose instanceCount, remove the Resource field and iterate on the exact shape that ModelConfig would take in the future in the design doc & future PRs.

Reply:
Yeah, I think it's better to remove it for now. I need to read through the docs again, but you can either run instances on CPU, or GPU for a model, I don't think you can choose both. 

I'm thinking that maybe a completely separate message for each model config using `oneof` will make it much more flexible, instead of trying to shoehorn it all into one succint message.

Reply:
You can choose both: https://github.com/triton-inference-server/python_backend/blob/f3068c03ed82e099cef5e2b40e9d1d79b1eab7ac/examples/instance_kind/README.md